File: frontend/lib/models/showroom.dart
class Showroom {
  final int id;
  final String name;
  final String? location;
  final bool isActive;
  final String? createdAt;

  Showroom({
    required this.id,
    required this.name,
    this.location,
    required this.isActive,
    this.createdAt,
  });

  factory Showroom.fromJson(Map<String, dynamic> json) {
    return Showroom(
      id: json['id'] as int,
      name: json['name'] as String,
      location: json['location'] as String?,
      isActive: json['is_active'] == true || json['is_active'] == 1,
      createdAt: json['created_at'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'location': location,
        'is_active': isActive,
      };
}

File: ios/Runner/Models/Showroom.swift
import Foundation

struct Showroom: Codable, Identifiable, Hashable {
    let id: Int
    let name: String
    let location: String?
    let isActive: Bool
    let createdAt: String?

    enum CodingKeys: String, CodingKey {
        case id, name, location
        case isActive  = "is_active"
        case createdAt = "created_at"
    }

    init(from decoder: Decoder) throws {
        let c = try decoder.container(keyedBy: CodingKeys.self)
        id       = try c.decode(Int.self, forKey: .id)
        name     = try c.decode(String.self, forKey: .name)
        location = try? c.decode(String.self, forKey: .location)
        if let b = try? c.decode(Bool.self, forKey: .isActive) {
            isActive = b
        } else {
            isActive = (try? c.decode(Int.self, forKey: .isActive)) == 1
        }
        createdAt = try? c.decode(String.self, forKey: .createdAt)
    }
}
